docs/050-breaking-changes.rst

```diff
@@ -272,6 +273,83 @@ Syntax
   ``override``, ``partial``, ``promise``, ``reference``, ``sealed``,
   ``sizeof``, ``supports``, ``typedef`` and ``unchecked``.
 
+.. _interoperability:
+
+Interoperability with older Contracts
+=====================================
+
+It is still possible to interface with contracts written for Solidity versions prior to
+v0.5.0 by defining interfaces for them. Consider you have the following pre-0.5.0
+contract already deployed:
+
+::
+
+   // This will not compile
+   pragma solidity ^0.4.25;
+   contract OldContract {
+      function someOldFunction(uint8 a) {
+         //...
+      }
+      function anotherOldFunction() constant returns (bool) {
+         //...
+      }
+      // ...
+   }
+
+This will no longer compile with Solidity v0.5.0. However, you can define a compatible interface for it:
+
+::
+
+   pragma solidity >0.4.99 <0.6.0;
+   interface OldContract {
+      function someOldFunction(uint8 a) external;
+      function anotherOldFunction() external returns (bool);
+   }
+
+Note that we did not declare ``anotherOldFunction`` to be ``view``, despite it being declared ``constant`` in the original
+contract. This is due to the fact that starting with Solidity v0.5.0 ``staticcall`` is used to call ``view`` functions.
+Prior to v0.5.0 the ``constant`` keyword was not enforced, so calling a function declared ``constant`` with ``staticcall``
+may still revert, since the ``constant`` function may still attempt to modify storage. Consequently, when defining an
+interface for older contracts, you should only use ``view`` in place of ``constant`` in case you are absolutely sure that
+the function will work with ``staticcall``.
+
+Given the interface defined above, you can now easily use the already deployed pre-0.5.0 contract:
+
+::
+
+   pragma solidity >0.4.99 <0.6.0;
+
+   interface OldContract {
+      function someOldFunction(uint8 a) external;
+      function anotherOldFunction() external returns (bool);
+   }
+
+   contract NewContract {
+      function doSomething(OldContract a) public returns (bool) {
+         a.someOldFunction(0x42);
+         return a.anotherOldFunction();
+      }
+   }
+
+Similarly, pre-0.5.0 libraries can be used by defining the functions of the library without implementation and
```

Comment:
I think might have approved this too early. Will check now, whether this is true ;)

Reply:
Yes, I just wrote something up rather quickly so far - we definitely should triple-check all of this :-).